src/test/pmemblk_create/pmemblk_create.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmemblk_create.c -- unit test for creating a block memory pool
+ *
+ * usage: pmemblk_create path bsize
+ */
+
+#include "unittest.h"
+
+#define	SIZEOF_NULLFILE_STR	(8)
+#define	SIZEOF_TESTFILE		(64 * 1024 * 1024)
+#define	CREATE_MODE			(0664)
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "pmemblk_create");
+
+	if (argc != 3)
+		FATAL("usage: %s path bsize", argv[0]);
+
+	PMEMblkpool *handle;
+	const char *path = argv[1];
+	size_t bsize = strtoul(argv[2], NULL, 0);
+
+	if (memcmp(path, "NULLFILE", SIZEOF_NULLFILE_STR)) {
```

Comment:
Yes, please get rid of the odd "NULLFILE" idea.  Since the first UNIX commands, having optional file arguments are quite commonplace.  Type "man cat" for example.

But looking at your code again, I think the NULLFILE code path is also strange because it then picks a specific path "./testfile".  I think a cleaner solution would be to always pass the path in, don't have one variation pick a fixed string for a path and another variation take the path from the command line.  Just always take the path from the command line and use another option if you want to force create versus open (like -c for create, -o for open).

In general, these are unit tests.  Nobody runs then except for the unit test scripts, so we do simple things like just check that we're given N args and expect them to be positional.  No problem.  So I'm inclined to give you quite a bit of freedom on how you define the CLI for a unit test.  But the NULLFILE thing is just too icky.  Please remove it.